packages/core/src/RenderPipeline/BasicRenderPipeline.ts

```diff
@@ -166,59 +100,100 @@ export class BasicRenderPipeline {
         this._updateMVPShaderData(context);
       }
     } else {
-      camera.shaderData.setTexture(Camera._cameraDepthTextureProperty, camera.engine._whiteTexture2D);
+      camera.shaderData.setTexture(Camera._cameraDepthTextureProperty, engine._whiteTexture2D);
+    }
+
+    // Check if need to create internal color texture
+    const independentCanvasEnabled = camera.independentCanvasEnabled;
+    if (independentCanvasEnabled && Vector4.equals(camera.viewport, PipelineUtils.defaultViewport)) {
+      console.warn(
+        "Camera use independent canvas and viewport cover the whole screen, it is recommended to disable antialias, depth and stencil to save memory when create engine."
+      );
     }
 
-    for (let i = 0, len = this._renderPassArray.length; i < len; i++) {
-      this._drawRenderPass(context, this._renderPassArray[i], camera, cubeFace, mipLevel, ignoreClear);
+    if (independentCanvasEnabled) {
+      const viewport = camera.pixelViewport;
+      const internalColorTarget = PipelineUtils.recreateRenderTargetIfNeeded(
+        engine,
+        this._internalColorTarget,
+        viewport.width,
+        viewport.height,
+        TextureFormat.R8G8B8A8,
+        TextureFormat.Depth24Stencil8,
```

Comment:
use `DepthStencil` in internal canvas is better, will use 32 depth format in webgl2

Reply:
Depth24Stencil8  is enough，DepthStencil may improve performance cost